js-api-spec/browser/compile.test.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {compileString, compileStringAsync} from 'sass';
+import type {OutputStyle} from 'sass';
+
+import {expectA} from './utils';
+
+describe('compileString', () => {
+  describe('success', () => {
+    describe('input', () => {
+      it('compiles SCSS by default', () => {
+        expect(compileString('$a: b; c {d: $a}').css).toBe('c {\n  d: b;\n}');
+      });
+
+      it('compiles SCSS with explicit syntax', () => {
+        expect(compileString('$a: b; c {d: $a}', {syntax: 'scss'}).css).toBe(
+          'c {\n  d: b;\n}'
+        );
+      });
+
+      it('compiles indented syntax with explicit syntax', () => {
+        expect(compileString('a\n  b: c', {syntax: 'indented'}).css).toBe(
+          'a {\n  b: c;\n}'
+        );
+      });
+
+      it('compiles plain CSS with explicit syntax', () => {
+        expect(compileString('a {b: c}', {syntax: 'css'}).css).toBe(
+          'a {\n  b: c;\n}'
+        );
+      });
+    });
+
+    it('recognizes the expanded output style', () => {
+      expect(compileString('a {b: c}', {style: 'expanded'}).css).toBe(
+        'a {\n  b: c;\n}'
+      );
+    });
+
+    describe('sourceMap', () => {
+      it("doesn't include one by default", () => {
+        expect(compileString('a {b: c}')).not.toHaveProperty('sourceMap');
+      });
+
+      it('includes one if sourceMap is true', () => {
+        const result = compileString('a {b: c}', {sourceMap: true});
+        expect(result).toHaveProperty('sourceMap');
+
+        // Explicitly don't test the details of the source map, because
+        // individual implementations are allowed to generate a custom map.
+        const sourceMap = result.sourceMap!;
+        expect(typeof sourceMap.version).toBeString();
+        expect(sourceMap.sources).toBeArray();
+        expect(sourceMap.names).toBeArray();
+        expect(sourceMap.mappings).toBeString();
+      });
+
+      it('includes one with source content if sourceMapIncludeSources is true', () => {
+        const result = compileString('a {b: c}', {
+          sourceMap: true,
+          sourceMapIncludeSources: true,
+        });
+        expect(result).toHaveProperty('sourceMap');
+
+        const sourceMap = result.sourceMap!;
+        expect(sourceMap).toHaveProperty('sourcesContent');
+        expect(sourceMap.sourcesContent!).toBeArray();
+        expect(sourceMap.sourcesContent!.length).toBeGreaterThanOrEqual(1);
+      });
```

Comment:
Of the tests I've ported over so far, these are the only that are [failing](https://github.com/sass/sass-spec/actions/runs/4767106037/jobs/8474966139?pr=1902):

```
dartException: TypeError: Instance of 'MappedListIterable<String, dynamic>': type 'MappedListIterable<String, dynamic>' is not a subtype of type 'Iterable<dynamic>'
```

@jerivas Do we want to support sourcemaps in the browser context? If so, could you add some tests for that in our `browser_test.dart` file?

Reply:
@nex3 I [added tests in Dart Sass](https://github.com/oddbird/dart-sass/commit/773ed19c32f54fc8b329c1d433fa99cb331df6f6) for sourceMaps, and didn't hit any errors. Any ideas why this test is still throwing an error when run in a browser context? It looks like the error starts with the call to `jsify(sourceMap)` in `node/compile.dart` here: https://github.com/sass/dart-sass/pull/1895/files#diff-5da2f39b0add2ab3649136ea6b10ff7c87690d2f895539c10511e5425c224024R164

Reply:
@nex3 Digging into this a bit more, I'm thinking it might be a bug in the dart2js `jsify` implementation, specifically [where `_convertDataTree` maps an `Iterable`](https://github.com/dart-lang/sdk/blob/main/sdk/lib/_internal/js_runtime/lib/js_patch.dart#L157). If I change that line from:

```dart
convertedList.addAll(o.map(_convert));
```

to:

```dart
convertedList.addAll(o.map(_convert).toList());
```

then the error goes away and the tests seem to pass in both contexts.

Should I file an issue?

Reply:
Yeah, I think filing that issue is a good idea. You could even send them a pull request for it.

Reply:
Filed an [issue](https://github.com/dart-lang/sdk/issues/52200), and skipped these tests in browser context temporarily. (I could submit a PR, but got bogged down trying to get Dart SDK tests to run locally, so decided to wait on that for now.)

Reply:
Resolved in https://github.com/oddbird/sass-spec/commit/9cca6820c579d66cbfa60788ce951c9fce89f574